TEST-#2008: Reduce runtime of CI checks a lot

vnlitvinov opened the pull request:
<!--
Thank you for your contribution!
Please review the contributing docs: https://modin.readthedocs.io/en/latest/development/contributing.html
if you have questions about contributing.
-->

## What do these changes do?

<!-- Please give a short brief about these changes. -->

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` <!-- you can amend your commit with a signature via `git commit -amend -s` -->
- [x] Resolves #2008 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date <!-- if you have added, renamed or removed files or directories please update the documentation accordingly -->

**Note**: this leaves `push` checks as they are (i.e. running to the full extent), only lessening the load for PRs, so we should pay attention to checks running after merging to `master` when this is in place.

anmyachev commented on .github/workflows/ci.yml:
```diff
@@ -653,6 +727,7 @@ jobs:
 
   upload-coverage:
     needs: [test-internals, test-api-and-no-engine, test-defaults, test-hdk, test-all-unidist, test-all, test-experimental, test-cloud]
+    if: github.event_name == 'push'
```
Will we not see the coverage in the pull request itself? Annotations were helpful.

vnlitvinov replied:
I think they won't be useful in PRs when this one is merged, as the tests being run would be different from run to run, and coverage percentage would be jumping wildly...

anmyachev replied:
We can disable the percentage coverage check (in the PR), but still see rows that are not covered directly in the pull request.

vnlitvinov replied:
That said, I don't see any place where we even set this threshold now... was it removed earlier?

anmyachev replied:
I don't remember exactly, but it looks like that.

vnlitvinov commented:
Stats on `test_io` on Windows - one of the longest in sanity subset (showing 90% of total time).

Proposal: drop `test_to_csv`, `test_read_parquet_directory` and `test_read_csv_parse_dates` from sanity set.
<details><summary>ray: total time = 0:30:26.348061</summary>

|Test|Time|
|-|-|
|modin/pandas/test/test_io.py::TestCsv::test_to_csv|0:20:42.298218|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet_directory|0:04:40.896809|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parse_dates|0:00:46.147396|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet_s3|0:00:44.041481|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_file_format|0:00:36.240867|

</details>
If dropped, new time: 0:04:17.005638
<hr>

<details><summary>dask: total time = 0:16:38.606221</summary>

|Test|Time|
|-|-|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet_directory|0:06:01.113421|
|modin/pandas/test/test_io.py::TestCsv::test_to_csv|0:02:18.008694|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parse_dates|0:01:27.040057|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parsing_2|0:01:08.547866|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_file_format|0:00:53.176036|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_datetime|0:00:36.454682|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet_s3|0:00:35.345149|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_internal|0:00:22.394443|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_skiprows_corner_cases|0:00:18.041728|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_quoting|0:00:17.510084|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_compression|0:00:11.822094|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet|0:00:11.490134|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parsing_1|0:00:09.128132|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_nans_handling|0:00:08.905340|
|modin/pandas/test/test_io.py::TestJson::test_read_json_s3|0:00:07.810503|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_delimiters|0:00:05.986995|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_default_to_pandas |0:00:05.869468|
|modin/pandas/test/test_io.py::TestFwf::test_read_fwf_s3|0:00:04.895532|

</details>
If dropped, new time: 0:06:52.444049
<hr>

<details><summary>unidist: total time = 0:13:33.153683</summary>

|Test|Time|
|-|-|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet_directory|0:04:52.663271|
|modin/pandas/test/test_io.py::TestCsv::test_to_csv|0:02:00.875977|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parse_dates|0:01:27.600703|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parsing_2|0:00:55.122581|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_file_format|0:00:34.805940|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_datetime|0:00:27.593711|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet_s3|0:00:21.400931|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_skiprows_corner_cases|0:00:18.707654|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_internal|0:00:17.206708|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_quoting|0:00:12.463342|
|modin/pandas/test/test_io.py::TestParquet::test_read_parquet|0:00:09.263109|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_compression|0:00:09.052779|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parsing_1|0:00:06.704983|
|modin/pandas/test/test_io.py::TestJson::test_read_json_s3|0:00:06.163425|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_nans_handling|0:00:05.765663|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_delimiters|0:00:04.340598|
|modin/pandas/test/test_io.py::TestCsv::test_read_csv_parsing_3|0:00:03.803814|

</details>
If dropped, new time: 0:05:12.013732

anmyachev commented on .github/actions/use-ray-cluster/action.yml:
```diff
@@ -0,0 +1,23 @@
+inputs:
+  ray-memory:
+    description: "Ray store memory size"
+    required: true
```
```suggestion
    required: false
    default: 1000000000
```

vnlitvinov replied:
I wanted to keep the number for memory limit in one place, namely in `ci.yml`

anmyachev replied:
ok

vnlitvinov replied:
I have reverted the change, it isn't playing nicely with other tests for some reason, and isn't saving much of a copy-paste anyway 😞

anmyachev commented on .github/workflows/ci.yml:
```diff
@@ -458,22 +419,13 @@ jobs:
       - name: Limit ray memory
         run: echo "MODIN_MEMORY=1000000000" >> $GITHUB_ENV
```
Can we remove it at all since we will use `./.github/actions/use-ray-cluster`?

vnlitvinov replied:
They're executed under different conditions, this is for non-Windows while `use-ray-cluster` is for Windows

anmyachev commented on .github/workflows/ci.yml:
```diff
@@ -652,7 +729,8 @@ jobs:
   upload-coverage:
-    needs: [test-internals, test-api-and-no-engine, test-defaults, test-hdk, test-all-unidist, test-all, test-experimental, test-cloud]
+    needs: [test-internals, test-api-and-no-engine, test-defaults, test-hdk, test-all-unidist, test-all, test-experimental, test-cloud, test-sanity]
+    if: always()  # we need to run it regardless of some job being skipped, like in PR
```
Why?

vnlitvinov replied:
Because if I do not specify a condition, by default this job will get skipped if any of its `needs` are skipped, which is most likely happening in a PR (where `test-all-unidist` is skipped).

vnlitvinov commented:
@anmyachev @devin-petersohn do you have any idea how to control the coverage failing this PR?
Right now all tests pass, but coverage check fails:
![image](https://github.com/modin-project/modin/assets/45396231/49c91ec2-c8bc-4d7f-8b43-83bd5e31ddb1)

My only bulletproof solution so far is to completely skip uploading the coverage...

vnlitvinov commented:
Stats on `test_series.py` and `test_map_metadata.py` - for now longest in Dask sanity subset (tables are showing 90% of total time).

Proposal: drop the following tests from the sanity, at least for Dask and Unidist:
```
test_map_metadata.py::test_constructor_from_modin_series
test_map_metadata.py::test_convert_dtypes_single_partition
test_map_metadata.py::test_drop_duplicates
test_series.py::test_value_counts
test_series.py::test_fillna
test_series.py::test_searchsorted
test_series.py::test___round__
test_series.py::test___setitem__
test_series.py::test_resample
test_series.py::test_sum
test_map_metadata.py::test_clip
```

<details><summary>ray-ubuntu: total time = 0:06:13.240758</summary>

|Test|Time|
|-|-|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor_from_modin_series|0:01:01.022590|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_single_partition|0:00:46.647023|
|modin/pandas/test/test_series.py::test_value_counts|0:00:24.359950|
|modin/pandas/test/dataframe/test_map_metadata.py::test_drop_duplicates|0:00:24.354860|
|modin/pandas/test/test_series.py::test_apply_return_df|0:00:12.880676|
|modin/pandas/test/test_series.py::test_searchsorted|0:00:10.419157|
|modin/pandas/test/test_series.py::test_fillna|0:00:10.386106|
|modin/pandas/test/test_series.py::test___round__|0:00:08.541045|
|modin/pandas/test/test_series.py::test_sum|0:00:08.481156|
|modin/pandas/test/test_series.py::test___setitem__|0:00:08.419980|
|modin/pandas/test/test_series.py::test_resample|0:00:08.174555|
|modin/pandas/test/dataframe/test_map_metadata.py::test_clip|0:00:06.214740|
|modin/pandas/test/test_series.py::test_le|0:00:05.719737|
|modin/pandas/test/test_series.py::test_loc|0:00:05.314849|
|modin/pandas/test/test_series.py::test_pop|0:00:03.925824|
|modin/pandas/test/test_series.py::test_shift|0:00:03.914961|
|modin/pandas/test/test_series.py::test_plot|0:00:03.751891|
|modin/pandas/test/test_series.py::test_str_replace|0:00:03.221837|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap|0:00:03.097546|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_index|0:00:02.767588|
|modin/pandas/test/test_series.py::test_isnull|0:00:02.645408|
|modin/pandas/test/test_series.py::test_items|0:00:02.447678|
|modin/pandas/test/test_series.py::test_apply_text_func|0:00:02.406206|
|modin/pandas/test/test_series.py::test_drop_duplicates|0:00:02.048750|
|modin/pandas/test/test_series.py::test_get|0:00:02.002945|
|modin/pandas/test/test_series.py::test_ge|0:00:01.930613|
|modin/pandas/test/test_series.py::test_autocorr|0:00:01.912079|
|modin/pandas/test/test_series.py::test_sort_index|0:00:01.883982|
|modin/pandas/test/test_series.py::test_str_slice|0:00:01.813913|
|modin/pandas/test/test_series.py::test_dtype|0:00:01.543929|
|modin/pandas/test/test_series.py::test_dt|0:00:01.540985|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap_numeric|0:00:01.447301|
|modin/pandas/test/test_series.py::test_sort_values|0:00:01.392671|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor|0:00:01.370477|
|modin/pandas/test/dataframe/test_map_metadata.py::test_dropna|0:00:01.353793|
|modin/pandas/test/test_series.py::test_replace |0:00:01.341509|
|modin/pandas/test/test_series.py::test_str_slice_replace|0:00:01.305862|
|modin/pandas/test/test_series.py::test_repeat|0:00:01.218220|
|modin/pandas/test/test_series.py::test_str_rindex|0:00:01.108787|
|modin/pandas/test/test_series.py::test_str_index|0:00:01.108156|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_dtype_backend|0:00:01.039575|
|modin/pandas/test/test_series.py::test_diff|0:00:00.982993|
|modin/pandas/test/test_series.py::test_str_rfind|0:00:00.972522|
|modin/pandas/test/test_series.py::test_str_find|0:00:00.957936|
|modin/pandas/test/test_series.py::test_asof|0:00:00.948500|
|modin/pandas/test/test_series.py::test_str_split|0:00:00.944271|
|modin/pandas/test/test_series.py::test_rdiv|0:00:00.890856|
|modin/pandas/test/test_series.py::test_std|0:00:00.871535|
|modin/pandas/test/test_series.py::test_reset_index|0:00:00.858212|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_axis|0:00:00.850856|
|modin/pandas/test/dataframe/test_map_metadata.py::test_astype_category |0:00:00.838486|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert_loc|0:00:00.818636|
|modin/pandas/test/dataframe/test_map_metadata.py::test___round__ |0:00:00.806417|
|modin/pandas/test/test_series.py::test_var|0:00:00.803025|
|modin/pandas/test/test_series.py::test_reindex|0:00:00.794290|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_multiple_row_partitions |0:00:00.772455|
|modin/pandas/test/test_series.py::test_apply_numeric|0:00:00.758587|
|modin/pandas/test/test_series.py::test_head|0:00:00.755683|
|modin/pandas/test/test_series.py::test_aggregate|0:00:00.748312|
|modin/pandas/test/test_series.py::test_str_contains|0:00:00.746548|
|modin/pandas/test/test_series.py::test_apply|0:00:00.715431|
|modin/pandas/test/test_series.py::test_duplicated|0:00:00.709273|
|modin/pandas/test/test_series.py::test_agg|0:00:00.708763|
|modin/pandas/test/test_series.py::test_array|0:00:00.691544|
|modin/pandas/test/test_series.py::test_matmul|0:00:00.680976|
|modin/pandas/test/test_series.py::test___getitem_edge_cases|0:00:00.652509|
|modin/pandas/test/test_series.py::test_ne|0:00:00.636161|
|modin/pandas/test/test_series.py::test_equals|0:00:00.634247|
|modin/pandas/test/test_series.py::test_cumsum|0:00:00.625893|
|modin/pandas/test/dataframe/test_map_metadata.py::test_add_suffix|0:00:00.620973|
|modin/pandas/test/dataframe/test_map_metadata.py::test_get_dummies|0:00:00.609023|
|modin/pandas/test/test_series.py::test_clip_sequence|0:00:00.596834|
|modin/pandas/test/test_series.py::test_str_rsplit|0:00:00.589143|
|modin/pandas/test/test_series.py::test_map|0:00:00.567763|
|modin/pandas/test/test_series.py::test_transform|0:00:00.567590|
|modin/pandas/test/test_series.py::test_quantile|0:00:00.565974|
|modin/pandas/test/test_series.py::test___or__|0:00:00.558629|
|modin/pandas/test/dataframe/test_map_metadata.py::test_empty_df |0:00:00.547486|
|modin/pandas/test/test_series.py::test_asof_large|0:00:00.526455|
|modin/pandas/test/test_series.py::test___truediv__|0:00:00.514863|
|modin/pandas/test/test_series.py::test___add__|0:00:00.514514|
|modin/pandas/test/test_series.py::test___sub__|0:00:00.497310|
|modin/pandas/test/test_series.py::test_unique|0:00:00.483159|
|modin/pandas/test/test_series.py::test_ravel_category|0:00:00.480144|
|modin/pandas/test/test_series.py::test_describe|0:00:00.477641|
|modin/pandas/test/test_series.py::test_dot|0:00:00.454100|
|modin/pandas/test/test_series.py::test_str_encode|0:00:00.442700|
|modin/pandas/test/test_series.py::test_divide|0:00:00.438080|
|modin/pandas/test/test_series.py::test_str_decode|0:00:00.434229|
|modin/pandas/test/test_series.py::test_corr|0:00:00.432347|
|modin/pandas/test/test_series.py::test_clip_scalar|0:00:00.428525|
|modin/pandas/test/test_series.py::test_copy|0:00:00.415937|
|modin/pandas/test/test_series.py::test_str_pad|0:00:00.411847|
|modin/pandas/test/test_series.py::test_sem_float_nan_only|0:00:00.406294|
|modin/pandas/test/dataframe/test_map_metadata.py::test_transpose|0:00:00.388295|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert|0:00:00.387384|
|modin/pandas/test/test_series.py::test___mul__|0:00:00.386363|
|modin/pandas/test/test_series.py::test_at_time |0:00:00.386123|
|modin/pandas/test/dataframe/test_map_metadata.py::test_squeeze |0:00:00.385265|
|modin/pandas/test/test_series.py::test_swapaxes|0:00:00.383186|
|modin/pandas/test/dataframe/test_map_metadata.py::test_isnull|0:00:00.381766|
|modin/pandas/test/test_series.py::test_sub|0:00:00.361304|
|modin/pandas/test/dataframe/test_map_metadata.py::test_update|0:00:00.356520|
|modin/pandas/test/test_series.py::test_aggregate_numeric|0:00:00.356022|
|modin/pandas/test/test_series.py::test_cumprod|0:00:00.354187|
|modin/pandas/test/test_series.py::test_mul|0:00:00.352023|
|modin/pandas/test/test_series.py::test_count|0:00:00.349465|

</details>
If dropped, new time: 0:02:36.219596
<hr>

<details><summary>dask-ubuntu: total time = 0:21:56.821851</summary>

|Test|Time|
|-|-|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor_from_modin_series|0:05:20.075259|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_single_partition|0:02:09.295321|
|modin/pandas/test/dataframe/test_map_metadata.py::test_drop_duplicates|0:01:27.327196|
|modin/pandas/test/test_series.py::test_value_counts|0:01:22.651335|
|modin/pandas/test/test_series.py::test_fillna|0:00:45.612526|
|modin/pandas/test/test_series.py::test_searchsorted|0:00:36.775779|
|modin/pandas/test/test_series.py::test___round__|0:00:32.377025|
|modin/pandas/test/test_series.py::test___setitem__|0:00:32.330326|
|modin/pandas/test/test_series.py::test_resample|0:00:30.016117|
|modin/pandas/test/test_series.py::test_sum|0:00:25.057627|
|modin/pandas/test/dataframe/test_map_metadata.py::test_clip|0:00:23.439819|
|modin/pandas/test/test_series.py::test_le|0:00:17.943441|
|modin/pandas/test/test_series.py::test_loc|0:00:17.221500|
|modin/pandas/test/test_series.py::test_plot|0:00:13.343844|
|modin/pandas/test/test_series.py::test_pop|0:00:13.092090|
|modin/pandas/test/test_series.py::test_shift|0:00:11.975570|
|modin/pandas/test/test_series.py::test_str_replace|0:00:10.321552|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap|0:00:09.046411|
|modin/pandas/test/test_series.py::test_isnull|0:00:08.295521|
|modin/pandas/test/test_series.py::test_items|0:00:08.202140|
|modin/pandas/test/test_series.py::test_sort_index|0:00:07.957156|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_index|0:00:07.852551|
|modin/pandas/test/test_series.py::test_apply_return_df|0:00:07.753275|
|modin/pandas/test/test_series.py::test_drop_duplicates|0:00:07.528529|
|modin/pandas/test/test_series.py::test_ge|0:00:07.009879|
|modin/pandas/test/test_series.py::test_apply_text_func|0:00:06.976747|
|modin/pandas/test/test_series.py::test_autocorr|0:00:06.691010|
|modin/pandas/test/test_series.py::test_get|0:00:06.656629|
|modin/pandas/test/test_series.py::test_str_slice|0:00:05.779688|
|modin/pandas/test/dataframe/test_map_metadata.py::test_dropna|0:00:05.606133|
|modin/pandas/test/test_series.py::test_dt|0:00:05.451609|
|modin/pandas/test/test_series.py::test_dtype|0:00:05.314484|
|modin/pandas/test/test_series.py::test_replace |0:00:04.775231|
|modin/pandas/test/test_series.py::test_sort_values|0:00:04.735213|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap_numeric|0:00:04.353577|
|modin/pandas/test/test_series.py::test_str_slice_replace|0:00:04.118870|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor|0:00:04.018275|
|modin/pandas/test/test_series.py::test_diff|0:00:03.864703|
|modin/pandas/test/test_series.py::test_asof|0:00:03.842877|
|modin/pandas/test/test_series.py::test_rdiv|0:00:03.579880|
|modin/pandas/test/test_series.py::test_str_find|0:00:03.345748|
|modin/pandas/test/test_series.py::test_reindex|0:00:03.063863|
|modin/pandas/test/test_series.py::test_std|0:00:02.999613|
|modin/pandas/test/test_series.py::test_str_rfind|0:00:02.951041|
|modin/pandas/test/test_series.py::test_apply_numeric|0:00:02.918520|
|modin/pandas/test/test_series.py::test_var|0:00:02.817026|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert_loc|0:00:02.797360|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_axis|0:00:02.764551|
|modin/pandas/test/test_series.py::test_agg|0:00:02.759176|
|modin/pandas/test/test_series.py::test_aggregate|0:00:02.752357|
|modin/pandas/test/dataframe/test_map_metadata.py::test___round__ |0:00:02.702119|
|modin/pandas/test/test_series.py::test_apply|0:00:02.675228|
|modin/pandas/test/test_series.py::test_str_index|0:00:02.595772|
|modin/pandas/test/test_series.py::test_str_rindex|0:00:02.529552|
|modin/pandas/test/test_series.py::test_cumsum|0:00:02.524442|
|modin/pandas/test/test_series.py::test_duplicated|0:00:02.502270|
|modin/pandas/test/test_series.py::test_equals|0:00:02.473373|
|modin/pandas/test/test_series.py::test_str_contains|0:00:02.449571|
|modin/pandas/test/test_series.py::test_str_split|0:00:02.293284|
|modin/pandas/test/test_series.py::test_ravel_category|0:00:02.179549|
|modin/pandas/test/test_series.py::test_dot|0:00:02.151002|
|modin/pandas/test/test_series.py::test_map|0:00:02.102494|
|modin/pandas/test/test_series.py::test_quantile|0:00:02.025412|
|modin/pandas/test/test_series.py::test_transform|0:00:02.009798|
|modin/pandas/test/test_series.py::test_clip_sequence|0:00:01.998991|
|modin/pandas/test/test_series.py::test_unique|0:00:01.975067|
|modin/pandas/test/test_series.py::test_asof_large|0:00:01.937110|
|modin/pandas/test/test_series.py::test___sub__|0:00:01.922489|
|modin/pandas/test/test_series.py::test___add__|0:00:01.871751|
|modin/pandas/test/test_series.py::test_describe|0:00:01.867307|
|modin/pandas/test/test_series.py::test___getitem_edge_cases|0:00:01.839605|
|modin/pandas/test/dataframe/test_map_metadata.py::test_get_dummies|0:00:01.838500|
|modin/pandas/test/test_series.py::test___truediv__|0:00:01.835734|
|modin/pandas/test/test_series.py::test_divide|0:00:01.791827|
|modin/pandas/test/test_series.py::test___or__|0:00:01.746573|
|modin/pandas/test/test_series.py::test_str_rsplit|0:00:01.706917|
|modin/pandas/test/test_series.py::test_matmul|0:00:01.702842|
|modin/pandas/test/test_series.py::test_head|0:00:01.685743|
|modin/pandas/test/test_series.py::test_clip_scalar|0:00:01.609136|
|modin/pandas/test/test_series.py::test_lt|0:00:01.574716|
|modin/pandas/test/test_series.py::test_reset_index|0:00:01.566283|
|modin/pandas/test/test_series.py::test___mul__|0:00:01.559576|
|modin/pandas/test/test_series.py::test___mod__|0:00:01.403282|
|modin/pandas/test/test_series.py::test_copy|0:00:01.377530|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert|0:00:01.364723|
|modin/pandas/test/test_series.py::test_corr|0:00:01.360599|
|modin/pandas/test/test_series.py::test_ffill|0:00:01.330398|
|modin/pandas/test/dataframe/test_map_metadata.py::test_squeeze |0:00:01.325499|
|modin/pandas/test/test_series.py::test_str_decode|0:00:01.324785|
|modin/pandas/test/test_series.py::test_aggregate_numeric|0:00:01.322111|
|modin/pandas/test/test_series.py::test_str_encode|0:00:01.316937|
|modin/pandas/test/dataframe/test_map_metadata.py::test_transpose|0:00:01.311942|
|modin/pandas/test/test_series.py::test_mul|0:00:01.301627|
|modin/pandas/test/test_series.py::test_cummax|0:00:01.300286|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_5653 |0:00:01.292354|
|modin/pandas/test/test_series.py::test_rsub|0:00:01.291822|
|modin/pandas/test/test_series.py::test_unstack|0:00:01.281707|
|modin/pandas/test/test_series.py::test_cumprod|0:00:01.275971|
|modin/pandas/test/test_series.py::test_sem_float_nan_only|0:00:01.275539|
|modin/pandas/test/dataframe/test_map_metadata.py::test_add_suffix|0:00:01.270708|

</details>
If dropped, new time: 0:07:51.863521
<hr>

<details><summary>unidist-ubuntu: total time = 0:05:57.655856</summary>

|Test|Time|
|-|-|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor_from_modin_series|0:01:13.874062|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_single_partition|0:00:30.971646|
|modin/pandas/test/test_series.py::test_value_counts|0:00:25.033282|
|modin/pandas/test/dataframe/test_map_metadata.py::test_drop_duplicates|0:00:22.658070|
|modin/pandas/test/test_series.py::test_fillna|0:00:11.528020|
|modin/pandas/test/test_series.py::test_searchsorted|0:00:10.321757|
|modin/pandas/test/test_series.py::test___round__|0:00:08.545259|
|modin/pandas/test/test_series.py::test___setitem__|0:00:08.385011|
|modin/pandas/test/test_series.py::test_resample|0:00:08.041218|
|modin/pandas/test/dataframe/test_map_metadata.py::test_clip|0:00:06.451728|
|modin/pandas/test/test_series.py::test_sum|0:00:06.128337|
|modin/pandas/test/test_series.py::test_le|0:00:05.951677|
|modin/pandas/test/test_series.py::test_apply_return_df|0:00:05.903710|
|modin/pandas/test/test_series.py::test_loc|0:00:05.899606|
|modin/pandas/test/test_series.py::test_pop|0:00:04.398418|
|modin/pandas/test/test_series.py::test_plot|0:00:04.309860|
|modin/pandas/test/test_series.py::test_shift|0:00:03.181661|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap|0:00:03.025574|
|modin/pandas/test/test_series.py::test_str_replace|0:00:02.733621|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_index|0:00:02.513644|
|modin/pandas/test/test_series.py::test_sort_index|0:00:02.419987|
|modin/pandas/test/test_series.py::test_items|0:00:02.224578|
|modin/pandas/test/test_series.py::test_drop_duplicates|0:00:02.189841|
|modin/pandas/test/test_series.py::test_ge|0:00:02.155987|
|modin/pandas/test/test_series.py::test_isnull|0:00:02.103575|
|modin/pandas/test/test_series.py::test_get|0:00:01.969222|
|modin/pandas/test/test_series.py::test_autocorr|0:00:01.835859|
|modin/pandas/test/test_series.py::test_apply_text_func|0:00:01.833704|
|modin/pandas/test/test_series.py::test_dt|0:00:01.738882|
|modin/pandas/test/test_series.py::test_str_slice|0:00:01.718644|
|modin/pandas/test/test_series.py::test_dtype|0:00:01.620798|
|modin/pandas/test/test_series.py::test_sort_values|0:00:01.616896|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap_numeric|0:00:01.547167|
|modin/pandas/test/dataframe/test_map_metadata.py::test_dropna|0:00:01.515749|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor|0:00:01.405483|
|modin/pandas/test/test_series.py::test_str_slice_replace|0:00:01.341021|
|modin/pandas/test/test_series.py::test_replace |0:00:01.297223|
|modin/pandas/test/test_series.py::test_diff|0:00:01.223816|
|modin/pandas/test/dataframe/test_map_metadata.py::test___round__ |0:00:01.044057|
|modin/pandas/test/test_series.py::test_asof|0:00:01.030881|
|modin/pandas/test/test_series.py::test_rdiv|0:00:00.976599|
|modin/pandas/test/test_series.py::test_str_rindex|0:00:00.921245|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_axis|0:00:00.895631|
|modin/pandas/test/test_series.py::test_reset_index|0:00:00.843278|
|modin/pandas/test/test_series.py::test_str_rfind|0:00:00.838671|
|modin/pandas/test/test_series.py::test_str_index|0:00:00.828172|
|modin/pandas/test/dataframe/test_map_metadata.py::test_astype_category |0:00:00.815029|
|modin/pandas/test/test_series.py::test_str_find|0:00:00.798161|
|modin/pandas/test/test_series.py::test_matmul|0:00:00.793378|
|modin/pandas/test/test_series.py::test_reindex|0:00:00.783372|
|modin/pandas/test/test_series.py::test_str_contains|0:00:00.767438|
|modin/pandas/test/test_series.py::test_duplicated|0:00:00.758626|
|modin/pandas/test/test_series.py::test_agg|0:00:00.750552|
|modin/pandas/test/test_series.py::test_std|0:00:00.744997|
|modin/pandas/test/test_series.py::test_aggregate|0:00:00.739516|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert_loc|0:00:00.737539|
|modin/pandas/test/test_series.py::test_str_split|0:00:00.731214|
|modin/pandas/test/test_series.py::test_head|0:00:00.702775|
|modin/pandas/test/test_series.py::test_cumsum|0:00:00.684663|
|modin/pandas/test/test_series.py::test_apply_numeric|0:00:00.680145|
|modin/pandas/test/test_series.py::test___getitem_edge_cases|0:00:00.667794|
|modin/pandas/test/test_series.py::test___sub__|0:00:00.658641|
|modin/pandas/test/test_series.py::test_apply|0:00:00.656160|
|modin/pandas/test/test_series.py::test_repeat|0:00:00.645962|
|modin/pandas/test/test_series.py::test_var|0:00:00.644224|
|modin/pandas/test/test_series.py::test_map|0:00:00.622127|
|modin/pandas/test/test_series.py::test_ne|0:00:00.622113|
|modin/pandas/test/test_series.py::test_clip_sequence|0:00:00.586861|
|modin/pandas/test/test_series.py::test_array|0:00:00.580739|
|modin/pandas/test/test_series.py::test___truediv__|0:00:00.566752|
|modin/pandas/test/test_series.py::test_describe|0:00:00.539153|
|modin/pandas/test/test_series.py::test_equals|0:00:00.537403|
|modin/pandas/test/test_series.py::test___or__|0:00:00.533916|
|modin/pandas/test/test_series.py::test___mul__|0:00:00.533548|
|modin/pandas/test/test_series.py::test_transform|0:00:00.532256|
|modin/pandas/test/test_series.py::test_asof_large|0:00:00.527490|
|modin/pandas/test/test_series.py::test_str_rsplit|0:00:00.525828|
|modin/pandas/test/test_series.py::test_ravel_simple_category|0:00:00.515885|
|modin/pandas/test/test_series.py::test___add__|0:00:00.514744|
|modin/pandas/test/test_series.py::test_divide|0:00:00.506665|
|modin/pandas/test/test_series.py::test_dot|0:00:00.506644|
|modin/pandas/test/test_series.py::test_clip_scalar|0:00:00.503947|
|modin/pandas/test/test_series.py::test_unique|0:00:00.503308|
|modin/pandas/test/test_series.py::test_quantile|0:00:00.502990|
|modin/pandas/test/test_series.py::test___mod__|0:00:00.490572|
|modin/pandas/test/test_series.py::test_ravel_category|0:00:00.467988|
|modin/pandas/test/dataframe/test_map_metadata.py::test_squeeze |0:00:00.437610|
|modin/pandas/test/test_series.py::test_mul|0:00:00.436479|
|modin/pandas/test/test_series.py::test_str_encode|0:00:00.435537|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert|0:00:00.427987|
|modin/pandas/test/test_series.py::test_rsub|0:00:00.419753|
|modin/pandas/test/dataframe/test_map_metadata.py::test_transpose|0:00:00.409883|
|modin/pandas/test/dataframe/test_map_metadata.py::test_get_dummies|0:00:00.402945|
|modin/pandas/test/test_series.py::test_str_decode|0:00:00.401784|
|modin/pandas/test/test_series.py::test_tail|0:00:00.401359|
|modin/pandas/test/test_series.py::test_rpow|0:00:00.396583|
|modin/pandas/test/test_series.py::test_lt|0:00:00.382830|
|modin/pandas/test/test_series.py::test_corr|0:00:00.382191|
|modin/pandas/test/test_series.py::test___neg__|0:00:00.378789|
|modin/pandas/test/test_series.py::test_at_time |0:00:00.373476|
|modin/pandas/test/test_series.py::test___pow__|0:00:00.373262|
|modin/pandas/test/dataframe/test_map_metadata.py::test_isnull|0:00:00.371732|
|modin/pandas/test/test_series.py::test___abs__|0:00:00.371256|
|modin/pandas/test/test_series.py::test_add_suffix|0:00:00.370552|
|modin/pandas/test/test_series.py::test_cummax|0:00:00.367157|
|modin/pandas/test/test_series.py::test_div|0:00:00.362165|
|modin/pandas/test/test_series.py::test___str__|0:00:00.355493|

</details>
If dropped, new time: 0:02:25.717466
<hr>

<details><summary>ray-windows: total time = 0:06:03.829523</summary>

|Test|Time|
|-|-|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor_from_modin_series|0:00:59.687313|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_single_partition|0:00:30.287464|
|modin/pandas/test/dataframe/test_map_metadata.py::test_drop_duplicates|0:00:22.875441|
|modin/pandas/test/test_series.py::test_apply_return_df|0:00:22.563695|
|modin/pandas/test/test_series.py::test_value_counts|0:00:22.370976|
|modin/pandas/test/test_series.py::test_searchsorted|0:00:12.981366|
|modin/pandas/test/test_series.py::test_fillna|0:00:10.543644|
|modin/pandas/test/test_series.py::test___round__|0:00:08.339877|
|modin/pandas/test/test_series.py::test___setitem__|0:00:08.304148|
|modin/pandas/test/test_series.py::test_sum|0:00:08.229378|
|modin/pandas/test/test_series.py::test_resample|0:00:07.360597|
|modin/pandas/test/test_series.py::test_head|0:00:06.257114|
|modin/pandas/test/dataframe/test_map_metadata.py::test_clip|0:00:05.704027|
|modin/pandas/test/test_series.py::test_le|0:00:05.355895|
|modin/pandas/test/test_series.py::test_loc|0:00:05.235567|
|modin/pandas/test/test_series.py::test_pop|0:00:03.865095|
|modin/pandas/test/test_series.py::test_plot|0:00:03.725157|
|modin/pandas/test/test_series.py::test_str_replace|0:00:03.101793|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap|0:00:03.041490|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_index|0:00:02.654596|
|modin/pandas/test/test_series.py::test_shift|0:00:02.511028|
|modin/pandas/test/test_series.py::test_isnull|0:00:02.346564|
|modin/pandas/test/test_series.py::test_apply_text_func|0:00:02.335983|
|modin/pandas/test/dataframe/test_map_metadata.py::test_empty_df |0:00:02.319387|
|modin/pandas/test/test_series.py::test_items|0:00:02.276585|
|modin/pandas/test/test_series.py::test_drop_duplicates|0:00:02.048518|
|modin/pandas/test/test_series.py::test_str_split|0:00:02.012091|
|modin/pandas/test/test_series.py::test_get|0:00:01.977570|
|modin/pandas/test/test_series.py::test_ge|0:00:01.945257|
|modin/pandas/test/test_series.py::test_sort_index|0:00:01.855193|
|modin/pandas/test/test_series.py::test_str_slice|0:00:01.807235|
|modin/pandas/test/test_series.py::test_autocorr|0:00:01.792638|
|modin/pandas/test/test_series.py::test_dt|0:00:01.538183|
|modin/pandas/test/test_series.py::test_sort_values|0:00:01.496936|
|modin/pandas/test/test_series.py::test_dtype|0:00:01.469408|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap_numeric|0:00:01.442433|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor|0:00:01.369202|
|modin/pandas/test/test_series.py::test_repeat|0:00:01.306195|
|modin/pandas/test/dataframe/test_map_metadata.py::test_dropna|0:00:01.302531|
|modin/pandas/test/test_series.py::test_str_slice_replace|0:00:01.257437|
|modin/pandas/test/test_series.py::test_replace |0:00:01.182344|
|modin/pandas/test/test_series.py::test_str_rindex|0:00:01.172683|
|modin/pandas/test/test_series.py::test_var|0:00:01.154174|
|modin/pandas/test/test_series.py::test_str_index|0:00:01.057530|
|modin/pandas/test/test_series.py::test_diff|0:00:00.999939|
|modin/pandas/test/test_series.py::test_array|0:00:00.987194|
|modin/pandas/test/test_series.py::test_str_rfind|0:00:00.945804|
|modin/pandas/test/test_series.py::test_str_find|0:00:00.927129|
|modin/pandas/test/test_series.py::test_asof|0:00:00.914949|
|modin/pandas/test/dataframe/test_map_metadata.py::test_astype_category |0:00:00.888860|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_axis|0:00:00.883205|
|modin/pandas/test/test_series.py::test_std|0:00:00.845292|
|modin/pandas/test/test_series.py::test_reset_index|0:00:00.842017|
|modin/pandas/test/test_series.py::test_rdiv|0:00:00.839932|
|modin/pandas/test/dataframe/test_map_metadata.py::test___round__ |0:00:00.831728|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert_loc|0:00:00.811535|
|modin/pandas/test/test_series.py::test_reindex|0:00:00.747252|
|modin/pandas/test/test_series.py::test_apply|0:00:00.715416|
|modin/pandas/test/test_series.py::test_apply_numeric|0:00:00.714440|
|modin/pandas/test/test_series.py::test_aggregate|0:00:00.706512|
|modin/pandas/test/test_series.py::test_duplicated|0:00:00.703790|
|modin/pandas/test/test_series.py::test_agg|0:00:00.698245|
|modin/pandas/test/test_series.py::test_str_contains|0:00:00.664962|
|modin/pandas/test/test_series.py::test___getitem_edge_cases|0:00:00.657076|
|modin/pandas/test/test_series.py::test_matmul|0:00:00.640199|
|modin/pandas/test/test_series.py::test_cumsum|0:00:00.613256|
|modin/pandas/test/dataframe/test_map_metadata.py::test_get_dummies|0:00:00.611356|
|modin/pandas/test/test_series.py::test_ne|0:00:00.601176|
|modin/pandas/test/dataframe/test_map_metadata.py::test_add_suffix|0:00:00.582271|
|modin/pandas/test/test_series.py::test_transform|0:00:00.577282|
|modin/pandas/test/test_series.py::test_map|0:00:00.571970|
|modin/pandas/test/test_series.py::test_equals|0:00:00.563654|
|modin/pandas/test/test_series.py::test_quantile|0:00:00.540026|
|modin/pandas/test/test_series.py::test___truediv__|0:00:00.538899|
|modin/pandas/test/test_series.py::test_asof_large|0:00:00.528709|
|modin/pandas/test/test_series.py::test___add__|0:00:00.527952|
|modin/pandas/test/test_series.py::test___sub__|0:00:00.508024|
|modin/pandas/test/test_series.py::test___or__|0:00:00.507609|
|modin/pandas/test/test_series.py::test_ravel_category|0:00:00.477097|
|modin/pandas/test/test_series.py::test_unique|0:00:00.474500|
|modin/pandas/test/test_series.py::test_str_encode|0:00:00.465531|
|modin/pandas/test/test_series.py::test_str_decode|0:00:00.449975|
|modin/pandas/test/test_series.py::test_ffill|0:00:00.436297|
|modin/pandas/test/test_series.py::test_clip_sequence|0:00:00.435607|
|modin/pandas/test/test_series.py::test_dot|0:00:00.432215|
|modin/pandas/test/test_series.py::test_describe|0:00:00.428800|
|modin/pandas/test/test_series.py::test_divide|0:00:00.425895|
|modin/pandas/test/dataframe/test_map_metadata.py::test_squeeze |0:00:00.422358|
|modin/pandas/test/test_series.py::test_clip_scalar|0:00:00.421991|
|modin/pandas/test/test_series.py::test___mul__|0:00:00.400971|
|modin/pandas/test/test_series.py::test_str_pad|0:00:00.396056|
|modin/pandas/test/test_series.py::test_swapaxes|0:00:00.392432|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert|0:00:00.389228|
|modin/pandas/test/test_series.py::test_at_time |0:00:00.382519|
|modin/pandas/test/test_series.py::test_copy|0:00:00.379515|
|modin/pandas/test/dataframe/test_map_metadata.py::test_transpose|0:00:00.379332|
|modin/pandas/test/test_series.py::test_corr|0:00:00.379095|
|modin/pandas/test/test_series.py::test_rsub|0:00:00.372886|
|modin/pandas/test/test_series.py::test_sub|0:00:00.366528|
|modin/pandas/test/test_series.py::test___mod__|0:00:00.365389|
|modin/pandas/test/test_series.py::test_str_match|0:00:00.362464|
|modin/pandas/test/test_series.py::test___neg__|0:00:00.354405|

</details>
If dropped, new time: 0:02:47.145292
<hr>

<details><summary>dask-windows: total time = 0:28:33.707301</summary>

|Test|Time|
|-|-|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor_from_modin_series|0:05:51.069536|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_single_partition|0:03:04.759426|
|modin/pandas/test/dataframe/test_map_metadata.py::test_drop_duplicates|0:02:04.474945|
|modin/pandas/test/test_series.py::test_value_counts|0:01:50.170447|
|modin/pandas/test/test_series.py::test_fillna|0:01:03.520360|
|modin/pandas/test/test_series.py::test_searchsorted|0:00:49.253543|
|modin/pandas/test/test_series.py::test___round__|0:00:45.068190|
|modin/pandas/test/test_series.py::test___setitem__|0:00:44.979027|
|modin/pandas/test/test_series.py::test_resample|0:00:37.881055|
|modin/pandas/test/test_series.py::test_sum|0:00:34.657787|
|modin/pandas/test/dataframe/test_map_metadata.py::test_clip|0:00:33.512029|
|modin/pandas/test/test_series.py::test_le|0:00:21.383870|
|modin/pandas/test/test_series.py::test_loc|0:00:20.969438|
|modin/pandas/test/test_series.py::test_pop|0:00:16.172498|
|modin/pandas/test/test_series.py::test_plot|0:00:15.798480|
|modin/pandas/test/test_series.py::test_shift|0:00:15.132096|
|modin/pandas/test/test_series.py::test_apply_return_df|0:00:14.675764|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap|0:00:13.063385|
|modin/pandas/test/test_series.py::test_str_replace|0:00:12.531478|
|modin/pandas/test/test_series.py::test_head|0:00:12.402998|
|modin/pandas/test/test_series.py::test_isnull|0:00:11.226076|
|modin/pandas/test/test_series.py::test_items|0:00:10.882250|
|modin/pandas/test/test_series.py::test_sort_index|0:00:10.453247|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_index|0:00:09.788310|
|modin/pandas/test/test_series.py::test_drop_duplicates|0:00:09.699882|
|modin/pandas/test/test_series.py::test_autocorr|0:00:09.628751|
|modin/pandas/test/test_series.py::test_ge|0:00:09.565872|
|modin/pandas/test/test_series.py::test_get|0:00:09.200013|
|modin/pandas/test/test_series.py::test_apply_text_func|0:00:08.930094|
|modin/pandas/test/dataframe/test_map_metadata.py::test_dropna|0:00:08.053200|
|modin/pandas/test/test_series.py::test_str_slice|0:00:07.354208|
|modin/pandas/test/test_series.py::test_dt|0:00:07.212187|
|modin/pandas/test/test_series.py::test_dtype|0:00:06.925560|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap_numeric|0:00:06.502800|
|modin/pandas/test/test_series.py::test_replace |0:00:06.045771|
|modin/pandas/test/test_series.py::test_sort_values|0:00:05.956924|
|modin/pandas/test/test_series.py::test_diff|0:00:05.611758|
|modin/pandas/test/test_series.py::test_str_slice_replace|0:00:05.237139|
|modin/pandas/test/test_series.py::test_asof|0:00:05.110139|
|modin/pandas/test/test_series.py::test_rdiv|0:00:04.858360|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor|0:00:04.395597|
|modin/pandas/test/test_series.py::test_reindex|0:00:04.383420|
|modin/pandas/test/test_series.py::test_str_find|0:00:04.329036|
|modin/pandas/test/test_series.py::test_cumsum|0:00:04.035489|
|modin/pandas/test/test_series.py::test_std|0:00:04.004678|
|modin/pandas/test/test_series.py::test_str_rfind|0:00:03.807584|
|modin/pandas/test/test_series.py::test_var|0:00:03.646790|
|modin/pandas/test/test_series.py::test_apply_numeric|0:00:03.605581|
|modin/pandas/test/test_series.py::test_agg|0:00:03.597369|
|modin/pandas/test/test_series.py::test_apply|0:00:03.566173|
|modin/pandas/test/test_series.py::test_equals|0:00:03.526406|
|modin/pandas/test/test_series.py::test_aggregate|0:00:03.509871|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert_loc|0:00:03.478739|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_axis|0:00:03.457097|
|modin/pandas/test/test_series.py::test_duplicated|0:00:03.265278|
|modin/pandas/test/test_series.py::test_str_rindex|0:00:03.250848|
|modin/pandas/test/test_series.py::test_str_index|0:00:03.185043|
|modin/pandas/test/test_series.py::test_str_contains|0:00:03.080347|
|modin/pandas/test/test_series.py::test_ravel_category|0:00:02.983221|
|modin/pandas/test/dataframe/test_map_metadata.py::test___round__ |0:00:02.945512|
|modin/pandas/test/test_series.py::test_str_split|0:00:02.934482|
|modin/pandas/test/test_series.py::test_clip_sequence|0:00:02.824119|
|modin/pandas/test/test_series.py::test_map|0:00:02.819110|
|modin/pandas/test/test_series.py::test_describe|0:00:02.728689|
|modin/pandas/test/test_series.py::test___sub__|0:00:02.716289|
|modin/pandas/test/test_series.py::test_dot|0:00:02.706028|
|modin/pandas/test/test_series.py::test___truediv__|0:00:02.703408|
|modin/pandas/test/test_series.py::test_transform|0:00:02.632982|
|modin/pandas/test/test_series.py::test___add__|0:00:02.593523|
|modin/pandas/test/test_series.py::test_asof_large|0:00:02.579081|
|modin/pandas/test/test_series.py::test_quantile|0:00:02.568741|
|modin/pandas/test/test_series.py::test_divide|0:00:02.531595|
|modin/pandas/test/dataframe/test_map_metadata.py::test_get_dummies|0:00:02.512382|
|modin/pandas/test/test_series.py::test_unique|0:00:02.486912|
|modin/pandas/test/test_series.py::test___or__|0:00:02.464073|
|modin/pandas/test/test_series.py::test_reset_index|0:00:02.426508|
|modin/pandas/test/test_series.py::test_str_rsplit|0:00:02.318467|
|modin/pandas/test/test_series.py::test___getitem_edge_cases|0:00:02.286119|
|modin/pandas/test/test_series.py::test_clip_scalar|0:00:02.172086|
|modin/pandas/test/test_series.py::test_matmul|0:00:02.108683|
|modin/pandas/test/test_series.py::test_corr|0:00:01.954752|
|modin/pandas/test/test_series.py::test_copy|0:00:01.944615|
|modin/pandas/test/test_series.py::test___mul__|0:00:01.928246|
|modin/pandas/test/test_series.py::test_ravel_simple_category|0:00:01.918430|
|modin/pandas/test/test_series.py::test_at_time |0:00:01.911774|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert|0:00:01.882097|
|modin/pandas/test/dataframe/test_map_metadata.py::test_squeeze |0:00:01.857380|
|modin/pandas/test/test_series.py::test_cumprod|0:00:01.845191|
|modin/pandas/test/test_series.py::test_mul|0:00:01.833380|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_5653 |0:00:01.820547|
|modin/pandas/test/test_series.py::test_repeat|0:00:01.812791|
|modin/pandas/test/test_series.py::test_ffill|0:00:01.788827|
|modin/pandas/test/test_series.py::test_cummax|0:00:01.771416|
|modin/pandas/test/test_series.py::test_agg_numeric|0:00:01.764940|
|modin/pandas/test/test_series.py::test___mod__|0:00:01.761978|
|modin/pandas/test/test_series.py::test_cummin|0:00:01.756815|
|modin/pandas/test/test_series.py::test_rpow|0:00:01.754355|
|modin/pandas/test/test_series.py::test_rsub|0:00:01.746922|
|modin/pandas/test/dataframe/test_map_metadata.py::test_transpose|0:00:01.745269|
|modin/pandas/test/test_series.py::test_ne|0:00:01.721162|
|modin/pandas/test/test_series.py::test_aggregate_numeric|0:00:01.712812|
|modin/pandas/test/test_series.py::test_div|0:00:01.706177|

</details>
If dropped, new time: 0:10:34.360956
<hr>

<details><summary>unidist-windows: total time = 0:33:17.997092</summary>

|Test|Time|
|-|-|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor_from_modin_series|0:13:22.012377|
|modin/pandas/test/dataframe/test_map_metadata.py::test_convert_dtypes_single_partition|0:02:30.602567|
|modin/pandas/test/dataframe/test_map_metadata.py::test_drop_duplicates|0:02:12.888360|
|modin/pandas/test/test_series.py::test_value_counts|0:01:50.588324|
|modin/pandas/test/test_series.py::test_fillna|0:00:50.735330|
|modin/pandas/test/test_series.py::test_searchsorted|0:00:39.733017|
|modin/pandas/test/test_series.py::test_resample|0:00:31.820497|
|modin/pandas/test/test_series.py::test_le|0:00:29.893253|
|modin/pandas/test/test_series.py::test_loc|0:00:29.017604|
|modin/pandas/test/dataframe/test_map_metadata.py::test_clip|0:00:26.912500|
|modin/pandas/test/test_series.py::test___round__|0:00:24.473547|
|modin/pandas/test/test_series.py::test___setitem__|0:00:24.344710|
|modin/pandas/test/test_series.py::test_pop|0:00:17.285032|
|modin/pandas/test/test_series.py::test_sort_index|0:00:16.598300|
|modin/pandas/test/test_series.py::test_plot|0:00:16.496603|
|modin/pandas/test/test_series.py::test_sum|0:00:14.368595|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_index|0:00:13.309112|
|modin/pandas/test/test_series.py::test_shift|0:00:12.915195|
|modin/pandas/test/test_series.py::test_apply_return_df|0:00:11.294150|
|modin/pandas/test/dataframe/test_map_metadata.py::test_constructor|0:00:09.435776|
|modin/pandas/test/test_series.py::test_ge|0:00:09.311329|
|modin/pandas/test/test_series.py::test_get|0:00:09.295936|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap|0:00:09.099180|
|modin/pandas/test/test_series.py::test_str_replace|0:00:09.004396|
|modin/pandas/test/test_series.py::test_head|0:00:08.738203|
|modin/pandas/test/dataframe/test_map_metadata.py::test___round__ |0:00:08.476478|
|modin/pandas/test/test_series.py::test_isnull|0:00:08.349067|
|modin/pandas/test/test_series.py::test_items|0:00:08.243903|
|modin/pandas/test/test_series.py::test_dt|0:00:07.676801|
|modin/pandas/test/dataframe/test_map_metadata.py::test_dropna|0:00:07.473960|
|modin/pandas/test/test_series.py::test_drop_duplicates|0:00:07.158621|
|modin/pandas/test/test_series.py::test_dtype|0:00:07.106556|
|modin/pandas/test/test_series.py::test_sort_values|0:00:06.468181|
|modin/pandas/test/test_series.py::test_autocorr|0:00:05.924799|
|modin/pandas/test/test_series.py::test_apply_text_func|0:00:05.253428|
|modin/pandas/test/test_series.py::test_replace |0:00:05.067379|
|modin/pandas/test/test_series.py::test_diff|0:00:05.043315|
|modin/pandas/test/test_series.py::test_str_slice|0:00:05.036411|
|modin/pandas/test/test_series.py::test_rdiv|0:00:04.720926|
|modin/pandas/test/dataframe/test_map_metadata.py::test_set_axis|0:00:04.546053|
|modin/pandas/test/test_series.py::test_asof|0:00:04.441496|
|modin/pandas/test/dataframe/test_map_metadata.py::test_applymap_numeric|0:00:04.315241|
|modin/pandas/test/dataframe/test_map_metadata.py::test_insert_loc|0:00:04.094642|
|modin/pandas/test/test_series.py::test_reindex|0:00:03.900898|
|modin/pandas/test/test_series.py::test_str_slice_replace|0:00:03.658641|
|modin/pandas/test/test_series.py::test_cumsum|0:00:03.564942|
|modin/pandas/test/test_series.py::test___sub__|0:00:03.346486|
|modin/pandas/test/test_series.py::test_str_rindex|0:00:03.295887|
|modin/pandas/test/test_series.py::test_dot|0:00:03.177115|
|modin/pandas/test/test_series.py::test___truediv__|0:00:03.112055|
|modin/pandas/test/test_series.py::test_duplicated|0:00:02.990776|
|modin/pandas/test/test_series.py::test___add__|0:00:02.985658|
|modin/pandas/test/test_series.py::test_describe|0:00:02.941209|
|modin/pandas/test/test_series.py::test_divide|0:00:02.930041|
|modin/pandas/test/test_series.py::test_str_rfind|0:00:02.893434|
|modin/pandas/test/test_series.py::test_str_find|0:00:02.861600|
|modin/pandas/test/test_series.py::test_aggregate|0:00:02.797036|
|modin/pandas/test/test_series.py::test_agg|0:00:02.766040|
|modin/pandas/test/test_series.py::test_asof_large|0:00:02.738598|
|modin/pandas/test/test_series.py::test_clip_sequence|0:00:02.687752|
|modin/pandas/test/test_series.py::test_str_index|0:00:02.657584|
|modin/pandas/test/test_series.py::test_apply|0:00:02.640915|
|modin/pandas/test/test_series.py::test___mul__|0:00:02.580742|
|modin/pandas/test/test_series.py::test_apply_numeric|0:00:02.546998|
|modin/pandas/test/test_series.py::test_ravel_category|0:00:02.479232|
|modin/pandas/test/test_series.py::test_str_contains|0:00:02.439294|
|modin/pandas/test/test_series.py::test_map|0:00:02.408289|
|modin/pandas/test/test_series.py::test_reset_index|0:00:02.361194|
|modin/pandas/test/test_series.py::test_std|0:00:02.354581|
|modin/pandas/test/test_series.py::test___mod__|0:00:02.300511|
|modin/pandas/test/test_series.py::test___or__|0:00:02.243193|

</details>
If dropped, new time: 0:09:49.517268
<hr>

vnlitvinov commented:
<details><summary>Script to produce the stats above</summary>

Requires logs to be downloaded. Easiest way is to download the whole logs of CI run and unzip only necessary.

```python

import datetime
import glob
def extract(data, marker):
    return '\n'.join(l for l in data.splitlines() if marker in l)

def process(backend, data, drop, merge_tests=False):
    entries = []
    for idx, l in enumerate(data.splitlines()):
        if 'PASSED' not in l:
            continue
        first, _ = l.split('PASSED')
        stamp, name = first.split(None, 1)
        stamp = datetime.datetime.strptime(stamp[:26], '%Y-%m-%dT%H:%M:%S.%f')
        entries.append((stamp, name))

    tt = {}
    for idx, (stamp, name) in enumerate(entries[:-1]):
        if merge_tests:
            name = name.split('[')[0]
        tt[name] = tt.get(name, datetime.timedelta()) + entries[idx+1][0] - stamp

    total = sum(tt.values(), datetime.timedelta())
    print(f'<details><summary>{backend}: total time = {total}</summary>')

    #assert len(tt) == len(entries) - 1
    tt1 = sorted(tt.items(), key=lambda e: (e[1], e[0]), reverse=True)

    print('\n|Test|Time|\n|-|-|')
    shown = 0
    for name, dur in tt1:
        if shown >= 0.9 * total.total_seconds():
            break
        shown += dur.total_seconds()
        print(f'|{name}|{dur}|')
    print('\n</details>')
    cut = sum((dur for name, dur in tt1 if not any(name.endswith(d) for d in drop)), datetime.timedelta())
    print(f'If dropped, new time: {cut}\n<hr>\n')
    #print(tt1[:10])

#drop = 'TestCsv::test_to_csv TestParquet::test_read_parquet_directory TestCsv::test_read_csv_parse_dates'.split()
drop = '''test_map_metadata.py::test_constructor_from_modin_series
test_map_metadata.py::test_convert_dtypes_single_partition
test_map_metadata.py::test_drop_duplicates
test_series.py::test_value_counts
test_series.py::test_fillna
test_series.py::test_searchsorted
test_series.py::test___round__
test_series.py::test___setitem__
test_series.py::test_resample
test_series.py::test_sum
test_map_metadata.py::test_clip
'''.splitlines()

for fname in glob.glob('*_test*.txt'):
    plat = fname.split()[0].split('-')[1]
    eng = fname.split()[2].strip(',')
    with open(fname) as inp:
        raw = inp.read()
        data = '\n'.join(extract(raw, marker) for marker in ('/test_series.py::', '/test_map_metadata.py::'))
        process(f'{eng}-{plat}', data, [d for d in drop if d], merge_tests=True)
```
</details>

anmyachev commented:
> @anmyachev @devin-petersohn do you have any idea how to control the coverage failing this PR? Right now all tests pass, but coverage check fails: ![image](https://user-images.githubusercontent.com/45396231/252154910-49c91ec2-c8bc-4d7f-8b43-83bd5e31ddb1.png)
> 
> My only bulletproof solution so far is to completely skip uploading the coverage...

How about defining a `fail_under` option for a PR of, say, around 10%?https://coverage.readthedocs.io/en/latest/config.html#report-fail-under

anmyachev commented:
> Stats on `test_series.py` and `test_map_metadata.py` - for now longest in Dask sanity subset (tables are showing 90% of total time). **Note**: stats gathered on a _non-parallel_ run (in this PR those tests are run in _parallel_).
> 
> Proposal: drop the following tests from the sanity, at least for Dask and Unidist:
> 
> ```
> test_map_metadata.py::test_constructor_from_modin_series
> test_map_metadata.py::test_convert_dtypes_single_partition
> test_map_metadata.py::test_drop_duplicates
> test_series.py::test_value_counts
> test_series.py::test_fillna
> test_series.py::test_searchsorted
> test_series.py::test___round__
> test_series.py::test___setitem__
> test_series.py::test_resample
> test_series.py::test_sum
> test_map_metadata.py::test_clip
> ```
> 
> ray-ubuntu: total time = 0:06:13.240758
> If dropped, new time: 0:02:36.219596
> dask-ubuntu: total time = 0:21:56.821851
> If dropped, new time: 0:07:51.863521
> unidist-ubuntu: total time = 0:05:57.655856
> If dropped, new time: 0:02:25.717466
> ray-windows: total time = 0:06:03.829523
> If dropped, new time: 0:02:47.145292
> dask-windows: total time = 0:28:33.707301
> If dropped, new time: 0:10:34.360956
> unidist-windows: total time = 0:33:17.997092
> If dropped, new time: 0:09:49.517268

@vnlitvinov agree to drop them for Dask and Unidist.

BTW @YarShev why is Unidist so much slower on Windows? (33 min vs 6 min on ubuntu)

vnlitvinov commented:
> How about defining a `fail_under` option for a PR of, say, around 10%?https://coverage.readthedocs.io/en/latest/config.html#report-fail-under

The behaviour of that check is controlled by codecov.com, and I don't think there is anything which is adjustable "for PR only"...
Ref: https://docs.codecov.com/docs/commit-status

I've tried to set some values which are (hopefully) okay for us.

vnlitvinov commented:
Docs builds are failing due to https://github.com/modin-project/modin/issues/6375 - completely unrelated, so I think this PR could be merged.

anmyachev commented on README.md:
```diff
@@ -13,6 +13,7 @@
 <a href="https://pepy.tech/project/modin"><img src="https://static.pepy.tech/personalized-badge/modin?period=total&units=international_system&left_color=black&right_color=blue&left_text=Downloads" align="center"></a>
 <a href="https://codecov.io/gh/modin-project/modin"><img src="https://codecov.io/gh/modin-project/modin/branch/master/graph/badge.svg" align="center"/></a>
 <a href="https://github.com/modin-project/modin/actions"><img src="https://github.com/modin-project/modin/workflows/master/badge.svg" align="center"></a>
+<a href="https://github.com/modin-project/modin/actions/workflows/ci.yml"><img src="https://github.com/modin-project/modin/actions/workflows/ci.yml/badge.svg?branch=master" align="center"></a>
```
Why?

vnlitvinov replied:
As quite a few tests are removed from the Sanity subset, I think we need an obvious way to check for CI tests passing against `master` branch, hence a new badge.

anmyachev replied:
How is this badge different from the one above?

vnlitvinov replied:
The links are different. Upper one leads to all actions, and added one leads specifically to `ci` workflow status.

anmyachev reviewed (approved):
LGTM! Great result!
